Rename config for docker

ionphractal opened the pull request:
After changing the domains to xip.io, a local Docker development environment was unable to load images (failing with 401 unauthorized). That is because the JWT token audience uri was still set to localhost and hence the token was not recognized as valid.

The configuration package loads the config from NODE_ENV. And as there was no development config and Docker uses "development" I though I'd rename the config and extend it with the required parameters.

Does this break anything on your side? @appinteractive @roschaefer

Commit:
Rename config for docker

appinteractive commented:
This would break the local environment without docker. You might want to use development-docker.json as a filename instead.

ionphractal commented:
Hm so we might want to treat `NODE_ENV=development-docker` as `development` in code, right?

roschaefer commented:
@ionphractal great debugging! Could you use `config/default-docker.json` instead, please? We set `NODE_APP_INSTANCE=docker` in the [Dockerfile](https://github.com/Human-Connection/API/blob/develop/Dockerfile#L8). See the [node-config wiki](https://github.com/lorenwest/node-config/wiki/Configuration-Files) to learn how config files are loaded.

Apart from that :+1:

roschaefer commented on config/development.json:
```diff
@@ -0,0 +1,15 @@
+      "audience": "http://webapp.127.0.0.1.xip.io:3030"
+    }
+  },
+  "seeder": {
```
And maybe remove `seeder` part completely. It should be up to the user if he or she wants to seed the database in docker.

ionphractal replied:
changed what you suggested

Commit:
Rename docker default config

roschaefer committed:
Remove redundancy

roschaefer commented on docker-compose.override.yml:
```diff
@@ -4,6 +4,7 @@ services:
   api:
     environment:
       - NODE_ENV=development
+      - NODE_APP_INSTANCE=docker
```
Not strictly required as it is set in the `Dockerfile` @ionphractal

roschaefer committed:
Remove redundant `NODE_APP_INSTANCE`

While it is important to have it set in the docker container, I would not add it additionally in `docker-compose.yml` because other developer might believe, it is necessary to have it there.

roschaefer commented:
@appinteractive I restarted the build once, can you tell me what's wrong with codacy right now?

appinteractive commented:
@roschaefer codacy?

roschaefer commented:
It's failing the build.
